From c0039918888e9e2d0f8edd284c52b3ffaabbef5b Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Wed, 18 Oct 2017 21:01:07 +0200
Subject: [PATCH] SAPI-293 moray storage layer's putObject implementation
 doesn't handle errors correctly Reviewed by: Julien Gilli
 <julien.gilli@joyent.com> Approved by: Julien Gilli <julien.gilli@joyent.com>

---
 lib/server/stor/moray.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/server/stor/moray.js b/lib/server/stor/moray.js
index aaded7d..cb26743 100644
--- a/lib/server/stor/moray.js
+++ b/lib/server/stor/moray.js
@@ -356,6 +356,7 @@ function putObject(bucket, uuid, obj, opts, cb) {
         if (err) {
             log.error(err, 'failed to put object %s', uuid);
             cb(err);
+            return;
         }
 
         var record = {};
-- 
2.21.0

